spec/datadog/tracing/span_operation_spec.rb

```diff
@@ -995,6 +995,86 @@
       end
     end
   end
+
+  describe '#record_exception' do
+    subject(:record_exception) do
+      begin
+        raise error
+      rescue => e
+        @error_object = Datadog::Core::Error.build_from(e)
+        span_op.record_exception(
+          e,
+          attributes: attributes,
+          timestamp: timestamp,
+          escaped: escaped
+        )
+      end
+    end
+
+    let(:error) { StandardError.new('test error') }
+    let(:attributes) { nil }
+    let(:timestamp) { nil }
+    let(:escaped) { false }
+    let(:error_object) { @error_object }
+
+    it 'creates a span event with exception details' do
+      record_exception
+
+      expect(span_op.span_events.last).to have_attributes(
+        name: 'exception',
+        attributes: {
+          'exception.type' => error_object.type,
```

Comment:
Instead of delegating to `error_object`, prefer the exact value assertion in test. For example :

```
"exception.type" => "StandardError",  "exception.message" => "test error", 
 ```